[AY1920S1-CS2113T-F09-1] Duke$$$

AjeyAshok opened the pull request:
@priyan-coder
@Araavinds
@mudaafi
@sallysohyun
@AjeyAshok

Rishav1 commented on src/main/java/executor/command/CommandAddIncomeReceipt.java:
Make RHS a private static final default string.

Rishav1 commented on src/main/java/executor/command/CommandAddIncomeReceipt.java:
SLAP violation.

Rishav1 commented on src/main/java/executor/command/CommandDelete.java:
Line too long.

Rishav1 commented on src/main/java/executor/command/CommandDelete.java:
Line too long.

Rishav1 commented on src/main/java/executor/command/CommandHelp.java:
Javadocs missing.

Rishav1 commented on src/main/java/executor/command/CommandIncome.java:
This function, along with several others seems redundant across several classes.

Rishav1 commented on src/main/java/executor/command/CommandNewTask.java:
`checkS` is a bad naming convention. Could be improved.

Rishav1 commented on src/main/java/executor/command/CommandReminder.java:
Violation of SLAP

Rishav1 commented on src/main/java/executor/command/CommandSchedule.java:
Comment not following coding standard.

Rishav1 commented on src/main/java/executor/command/CommandSchedule.java:
Comment not following coding standard.

Rishav1 commented on src/main/java/executor/task/Deadline.java:
Make the string private static final class variable

Rishav1 commented on src/main/java/executor/task/Deadline.java:
Javadocs without function. bad.

Rishav1 commented on src/main/java/executor/task/Task.java:
uninformative comment.

Rishav1 commented on src/main/java/storage/Storage.java:
Violation of SLAP.

Rishav1 commented on src/main/java/storage/Storage.java:
Violation of SLAP

Rishav1 commented on src/test/java/CommandWalletTest.java:
Don't add commented code.

Rishav1 reviewed (commented):
Good work team!

Apart from some minor comments, the code looks really good! Well done!

Rishav1 commented:
Also take a look at the reviews I added in #76. Some of them are still relevant to this PR.

priyan-coder committed:
changed access modifiers

Commit:
Update CommandAdd.java

Commit:
Update CommandDiv.java

Commit:
Update CommandMul.java

Commit:
Update CommandSub.java

priyan-coder committed:
trying to make setbalance a one time implementation

AjeyAshok committed:
Made changes after reviewing comments

sallysohyun committed:
Merge pull request #4 from AY1920S1-CS2113T-F09-1/master

f

priyan-coder committed:
now setbalance can only be executed once

priyan-coder committed:
Merge pull request #212 from priyan-coder/add-exceptions-currencyconverter

Handled exceptions for Currency Converter to resolve issues #199 , #189 , #183 , #155 , #130

priyan-coder committed:
refactored Date to LocalDate in Task Class and Command Reminder

AjeyAshok committed:
Added expendday & checkstyle done

priyan-coder committed:
refactoring commandschedule

priyan-coder committed:
refactored command schedule

priyan-coder committed:
refactored command schedule and command reminder but now deadline shows errors

priyan-coder committed:
refactored deadline

priyan-coder committed:
refactored todo

AjeyAshok committed:
Added function to help to find explanation for specific commands by user and prevent invalid input by user

priyan-coder committed:
refactored reminder, schedule, todo, deadline to make the date and time formatting

AjeyAshok committed:
Merge pull request #217 from AjeyAshok/exception-for-expenditure

Added exceptions to expenditure

AjeyAshok committed:
Commiting new help

AjeyAshok committed:
Merge branch 'master' into AjeyAshok-change-help-formatting

AjeyAshok committed:
Merge pull request #225 from AjeyAshok/AjeyAshok-change-help-formatting

Added function to help command .

AjeyAshok committed:
Merge pull request #218 from Araavinds/b-calculator

Added simple arithmetic command line based functionalities

priyan-coder committed:
Merge branch 'master' into reminder-fix

AjeyAshok committed:
Merge branch 'master' into bug-fix-setbalance

AjeyAshok committed:
Merge pull request #226 from AjeyAshok/expend-day

Added expendday & checkstyle done

Commit:
Handled exceptions for taglist and datelist commands

Mudaafi committed:
Merge pull request #224 from priyan-coder/reminder-fix

Fixes date management for Todo, Task, Deadline, Deadline, Command Reminder, Command Schedule to resolve issues #222, #171, #153 , $150

Commit:
Did some exception handling for calculator functions

AjeyAshok committed:
Provides the current date for expendedday

AjeyAshok committed:
Merge pull request #229 from AjeyAshok/changess

Changes done to CommandGetSpendingByDay to output current date

Mudaafi committed:
Successfully revamped the Architecture.

Todo:
 - Revamp Tests to match
 - Amend Checkstyle Violations
 - Move TaskList, Wallet, ReceiptTracker, etc classes to Storage instead of Ui
Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
- Revamped Tests to match
 - Amended Checkstyle Violations

Todo:
 - Move TaskList, Wallet, ReceiptTracker, etc classes to Storage instead of Ui
Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Merged with upstream.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Commit:
Merge branch 'master' into b-exceptionsCal

Commit:
Merge pull request #228 from Araavinds/b-exceptionsCal

Handled exceptions for calculator functions

Commit:
Merge branch 'master' into b-exceptionsDateAndTag

Commit:
Merge pull request #227 from Araavinds/b-exceptionsDateAndTag

Handled exceptions for taglist and datelist commands

priyan-coder committed:
Merge pull request #221 from priyan-coder/bug-fix-setbalance

Made Setbalance to be set only once and cannot be set to negative amounts to resolve issues #196, #174

Mudaafi committed:
Merge branch 'master' into architecture-update

priyan-coder committed:
Merge pull request #230 from Mudaafi/architecture-update

Architecture update

priyan-coder committed:
fixed bugs after merging the changes

priyan-coder committed:
finished checkstyle errors

priyan-coder committed:
Merge pull request #232 from priyan-coder/bug-fixes

Fixed bugs after merging architectural changes

sallysohyun committed:
Merge pull request #5 from AY1920S1-CS2113T-F09-1/master

5

priyan-coder committed:
added jUnit and resolved checkstyle issues

Commit:
new command to delete receipts

priyan-coder committed:
refactored the spacing issues with CommandWeather

priyan-coder committed:
added description for weather and also tests pass

sallysohyun committed:
Tried fixing issue of receipt being created when there is no cash input

sallysohyun committed:
no message

priyan-coder committed:
fixed the jUnit tests and also finished checkstyle errors

priyan-coder committed:
Merge pull request #237 from priyan-coder/weather-jUnit

Added jUnit testing for CommandWeather to resolve issue #236

priyan-coder committed:
Merge pull request #233 from priyan-coder/reminder-jUnit

added jUnit and resolved checkstyle issues to resolve issue #231

Mudaafi committed:
Fixed bug where CommandConvert will exit the programme when incorrect input was entered.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

priyan-coder committed:
Merge pull request #239 from Mudaafi/bugfix-convert

Fixed bug where CommandConvert will exit the programme when incorrect…

Commit:
Merge pull request #234 from Araavinds/b-deleteReceipts

New command to delete receipts

Commit:
Added jUnit testing for datelist,taglist and calculator commands

Commit:
Checkstyle test done

Commit:
Did some additional testing

Mudaafi committed:
GUI For Duke$$$ (#223)

* Updating javafx dependencies for Graphical User Interface.
* Implements the following GUI Elements:
- Task List Display
- Balance Chart

* Fixed a bug in PieChart and Parser.
* Fixed a bug in displaying of tasks.
* Implemented the bare-bones breakdownChart (stacked bar chart)
* Added in breakdownChart functionality for GUI
* Soft integration of GUI with Duke$$$.
More bugs to be researched on.

* GUI Partial Integration.
Requires:
 - Command Line Output Emulator
 - Balance Figure
 - Hover Over Concept

* PieChart bug fixed.
CommandLineDisplay in development.

* Implemented toggling between screens
* Resolved Merge Conflicts
* Near-complete integration of GUI with existing Features.
* Quality of Life adjustments to the Toast Msgs
* Resolved Checkstyle Violations
* Added the Home and CLI Commands so that the user can navigate between the two.
* Fixed Bugs found in CommandConvert with GUI.
* Amended the GUI to be compatible with the new Architecture.

WARNING: GUI Pulls Wallet AND TaskList into its layer for execution.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

AjeyAshok committed:
Merge pull request #240 from Araavinds/b-Junit

Added jUnit testing for datelist,taglist and calculator commands

AjeyAshok committed:
Junit-testing done for Month and Year

Commit:
Merge pull request #242 from AjeyAshok/junit-testing

Junit-testing done for Month and Year

sallysohyun committed:
Merge pull request #6 from AY1920S1-CS2113T-F09-1/master

mbu

Mudaafi committed:
Fixed JUnit test for CommandDateListTest.

Amended Checkstyle Violations for Main Code.

TODO:
Amend Checkstyle Violations for Test Code.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Merge pull request #243 from Mudaafi/junit-fix-commanddatelisttag

Fixed JUnit test for CommandDateListTest.

Mudaafi committed:
Enhanced Toast Pop-up so it gives more time for the user to read the longer the message is.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Commit:
Added new feature to see percentage of expenses on a tag

AjeyAshok committed:
Fixed checkstyle issues with test files

AjeyAshok committed:
Merge pull request #248 from AjeyAshok/master

Fixed checkstyle issues with test files

Commit:
Merge pull request #245 from Mudaafi/enhance-toast-time

Enhanced Toast Pop-up so it gives more time for the user to read the …

Commit:
Made some changes in the command percent class

Commit:
Merge branch 'master' into b-percent

Commit:
Merge pull request #247 from Araavinds/b-percent

Added new feature to see percentage of wallet expenses on a particular tag

priyan-coder committed:
added testing for CommandConvert

priyan-coder committed:
Merge pull request #249 from priyan-coder/testing-currency

Tested Command Convert

sallysohyun committed:
Merge pull request #7 from AY1920S1-CS2113T-F09-1/master

jbbj

Mudaafi committed:
Fixed Gui bug

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Merge pull request #251 from Mudaafi/bugfix-gui-home-display

Fixed Gui bug

priyan-coder committed:
initial set up of budget is done

priyan-coder committed:
budget feature implemented successfully

priyan-coder committed:
adding percentage features to budget command

priyan-coder committed:
added the budget feature

priyan-coder committed:
feature works

priyan-coder committed:
Merge pull request #255 from priyan-coder/create-budget

Able to set budget and display percentage details based on expenditure to resolve issues #253 , #254

Mudaafi committed:
Added Tracking Functionality via the Breakdown Chart.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

AjeyAshok committed:
Feature to get Expenditure for week + checkstyle testing done

priyan-coder committed:
fixed the bug

Commit:
Handled exceptions and added jUnit testing for command receiptdelete

Commit:
minor changes

Commit:
Merge pull request #257 from Araavinds/b-additionalTest

Handled exceptions and added jUnit testing for command receiptdelete

priyan-coder committed:
fixed checstyle issues

priyan-coder committed:
Merge pull request #258 from priyan-coder/fix-bug-in-setbalance

Setbalance should be able to set only once hereby.

AjeyAshok committed:
Merge branch 'master' into week-expend

AjeyAshok committed:
Merge pull request #256 from AjeyAshok/week-expend

Feature to get Expenditure for week + checkstyle testing done

Mudaafi committed:
Updated

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Merge branch 'master' into new-command-track

AjeyAshok committed:
Compeleted Junit testing for spending by week

priyan-coder committed:
adding ability to export csv form of either task or wallet

priyan-coder committed:
adding the export csv

priyan-coder committed:
now data.csv is being created but buggy

priyan-coder committed:
kind of works with better capability but abit buggy

priyan-coder committed:
reformatted

Commit:
Fixed bugs for deletereceipt, datelist and percent commands

priyan-coder committed:
feature works

Commit:
Update CommandDelete.java

Commit:
Merge pull request #267 from Araavinds/b-fixbugs

Fixed bugs for deletereceipt, datelist and percent commands

Commit:
Merge branch 'master' into week-Junit

Commit:
Merge pull request #264 from AjeyAshok/week-Junit

Compeleted Junit testing for spending by week

Commit:
Fixed bug for command delete

Commit:
minor change

Commit:
New feature that shows major expenses

Commit:
updated the feature to show expenditures above/equal to $100 upon majorexpense command without any primary input

priyan-coder committed:
removed the data.csv

priyan-coder committed:
Merge branch 'master' into implement-export-csv

priyan-coder committed:
Merge pull request #270 from priyan-coder/implement-export-csv

Implemented export csv which simply convert txt files to csv

priyan-coder committed:
able to generate csv but it looks kind of weird

priyan-coder committed:
refactor works succesfully

priyan-coder committed:
some help needed for the string parsing as its abit buggy

Commit:
Merge branch 'master' into b-fixDelete

Commit:
Merge pull request #268 from Araavinds/b-fixDelete

Fixed bug for command delete

Commit:
Added more jUnit testing for commands

Commit:
Merge pull request #272 from Araavinds/b-addTest

Added more jUnit testing for commands

priyan-coder committed:
managed to fix the bugs and now works fine

priyan-coder committed:
fixed some more bugs with export command

priyan-coder committed:
done with checkstyle and feature is successful

priyan-coder committed:
Merge branch 'master' into refactor-export

priyan-coder committed:
Merge pull request #273 from priyan-coder/refactor-export

Refactored  export Command successfully to be able to export wallet data into a useful display in excel and resolves issue #271

AjeyAshok committed:
Optimised CommandHelp function and added Junit testing

AjeyAshok committed:
Merge branch 'master' into bug-fix

Commit:
some minor changes

Commit:
Merge branch 'master' into b-majorExpense

Commit:
Merge pull request #269 from Araavinds/b-majorExpense

New feature that shows major expenses in the wallet

Mudaafi committed:
Implemented CommandTrack and Untrack

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Resolved Merge Conflicts

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Added Storage Support

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Merge pull request #261 from Mudaafi/new-command-track

Implements CommandTrack feature which allows User to track expenditure based on <tag>

priyan-coder committed:
fixed the bug

Commit:
txt

Commit:
t

AjeyAshok committed:
Merge branch 'master' into bug-fix

Commit:
Added new feature that shows statistics for a tag

Commit:
some changes

Commit:
Tested command Delete

Commit:
Merge pull request #276 from Araavinds/b-deleteTest

Tested Command Delete

AjeyAshok committed:
Merge pull request #274 from AjeyAshok/bug-fix

Optimised CommandHelp function and added Junit testing (DONT MERGE yet)

AjeyAshok committed:
New feature CommandEdit added with Junit testing

AjeyAshok committed:
Added documentation for funcitons in CommandHelp

priyan-coder committed:
fixed and styled

priyan-coder committed:
Merge pull request #278 from priyan-coder/fixbug-budget

Fixbug budget

priyan-coder committed:
done with test and redid a failed weather test

priyan-coder committed:
Merge pull request #279 from priyan-coder/setbalance-junit

Tested setbalance and also resolved a failed commandWeather test

priyan-coder committed:
done with export test

Mudaafi committed:
Adds clearCli Command which clears the text currently displayed on the cli.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Added singular JUnit test and amended any checkstyle violations.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

priyan-coder committed:
Merge pull request #282 from priyan-coder/test-exportCommand

Export command test completed

priyan-coder committed:
added testing for budget

priyan-coder committed:
Merge pull request #283 from priyan-coder/test-budget

added testing for budget

Commit:
Merge pull request #275 from Araavinds/b-stats

New Command Statistics feature that shows stats for a particular tag

priyan-coder committed:
added mock data in savedWallet

priyan-coder committed:
Merge pull request #284 from priyan-coder/add-mock-data

added mock data in savedWallet

Mudaafi committed:
Merge pull request #280 from Mudaafi/new-command-clear

Adds CommandClearCli which allows the User to clear the current CLI Display

AjeyAshok committed:
Merge branch 'master' into CommandEdit

Commit:
Updated Command percent,taglist and statistics to work for expenditures only

Commit:
Merge pull request #285 from Araavinds/b-updatePercent

Updated command percent,taglist and statistics to work for expenses only

priyan-coder committed:
removed setbalance and refactored

Mudaafi committed:
Adds tester support via CommandEnableTest which loads tester data into the session.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

priyan-coder committed:
Merge branch 'master' into remove-setbalance

Mudaafi committed:
Fixed Merge Conflicts.

Mudaafi committed:
Merge pull request #288 from Mudaafi/tester-aid

Allows testers to load pre-set data from app

Mudaafi committed:
Merge pull request #287 from priyan-coder/remove-setbalance

Removed setbalance and refactored code.

Mudaafi committed:
Fixes errors arising from the refactoring of Receipts to Income and Expenses.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Merge pull request #293 from Mudaafi/bugfix-balance-refactor

Fixes errors arising from the refactoring of Receipts to Income and E…

Mudaafi committed:
Additional Bug found where some methods in ReceiptTracker which return another ReceiptTracker are not initialized properly.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Additional Bug found where some methods in ReceiptTracker which return another ReceiptTracker are not initialized properly.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Merge pull request #294 from Mudaafi/bugfix-balance-refactor

Bugfix balance refactor

AjeyAshok committed:
CommandEdit feature optimised.
Pending optimisation for CommandHelp and CommandExpendedDay

Signed-off-by: AjeyAshok <a5000shiok@gmail.com>

AjeyAshok committed:
CommandEdit feature optimised.
Pending optimisation for CommandHelp and CommandExpendedDay

Signed-off-by: AjeyAshok <a5000shiok@gmail.com>

AjeyAshok committed:
CommandEdit feature optimised.
Pending optimisation for CommandHelp and CommandExpendedDay

Signed-off-by: AjeyAshok <a5000shiok@gmail.com>

AjeyAshok committed:
new adds

priyan-coder committed:
fixed reminder and viewschedule bugs

priyan-coder committed:
added back wallet datta

priyan-coder committed:
fixed checkstyle

Commit:
Optimized command majorexpense to no longer print empty list for inputs that do not satisfy condition

Commit:
Merge pull request #298 from Araavinds/b-optimize

Optimized command majorexpense

Commit:
removed command percent

Commit:
Merge pull request #299 from Araavinds/b-remove

Removed command percent

AjeyAshok committed:
CommandExpendedByDay rectified

Commit:
added testing for command stats and majorexpense

Commit:
Merge pull request #300 from Araavinds/b-optstats

added testing for command stats and majorexpense

AjeyAshok committed:
Junit testing added to CommandSpendingByDay

Commit:
tried optimizing taglists

AjeyAshok committed:
CommandHelp, CommandEdit, CommandExpended optimised

priyan-coder committed:
Merge branch 'master' into fix-reminder_bugsss

priyan-coder committed:
Merge pull request #297 from priyan-coder/fix-reminder_bugsss

Managed to fix reminder and viewschedule related bugs

priyan-coder committed:
updated wallet data which was missing

priyan-coder committed:
Merge pull request #302 from priyan-coder/update-data

updated wallet data which was missing

priyan-coder committed:
made command budget testing more detailed

priyan-coder committed:
added back the signs

priyan-coder committed:
Merge pull request #305 from priyan-coder/add-more-testing

made command budget testing more detailed

Mudaafi committed:
Merge branch 'master' into master

Mudaafi committed:
Merge pull request #250 from sallysohyun/master

Fixed receipt being created when there is no cash value input

Mudaafi committed:
Removed random println.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Merge branch 'master' into tester-aid

Mudaafi committed:
Fixed bug that did not allow Deadline to load from storage. Bug arose from bugfix involving Tasks, specifically Deadline.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Merge pull request #306 from Mudaafi/tester-aid

Tester aid

Mudaafi committed:
Fixed bug that did not allow Help command to properly show descriptions for creating Tasks.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Removed leftover code that prints help to CLI as well.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Merge pull request #307 from Mudaafi/bugfix-help-tasks

Fixed bug that did not allow Help command to properly show descriptio…

Commit:
Optimized commands taglist,datelist and stats

Commit:
removed unnecessary duke exceptions from receiptTracker test

Commit:
Merge branch 'master' into b-optTag

Commit:
Merge pull request #308 from Araavinds/b-optTag

Optimized command dateList, tagList and statistics

priyan-coder committed:
testtingggg

Commit:
Added more extensive testing to Command Stats

Commit:
Merge pull request #310 from Araavinds/b-moreStats-Test

Added more extensive testing to Command Stats

priyan-coder committed:
need help with testing export

AjeyAshok committed:
Rectified issue on multipple flag in commandEdit

priyan-coder committed:
done but cant seem to get the exporter to detect

priyan-coder committed:
data

priyan-coder committed:
Update testWalletDataSave.txt

priyan-coder committed:
Update savedWallet.txt

priyan-coder committed:
Merge pull request #311 from priyan-coder/tester-exporter

Tester exporter

Commit:
Removed taglist

Commit:
Added more extensive testing to Command Stats

Commit:
Merge pull request #313 from Araavinds/b-RemoveTag

Removed taglist

Mudaafi committed:
Merge branch 'master' into CommandEdit

AjeyAshok committed:
Final changes

AjeyAshok committed:
Merge branch 'CommandEdit' of https://github.com/AjeyAshok/main into CommandEdit

AjeyAshok committed:
All test cases rechecked

Mudaafi committed:
Merge pull request #277 from AjeyAshok/CommandEdit

New feature CommandEdit added with Junit testing

Mudaafi committed:
Made some minor bugfixes. (#314)

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Moves Java Classes to the layers they're supposed to be in (#315)

* Made some minor bugfixes.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

* Moved Classes to their proper places.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Moved Stats Window up (#316)

* Made some minor bugfixes.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

* Fixed height issue.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Update README.adoc

Mudaafi committed:
Update README.adoc

Mudaafi committed:
Delete README.md

Mudaafi committed:
Update AboutUs.adoc

Mudaafi committed:
Update AboutUs.adoc

Commit:
code optimised and added some testing

Commit:
Merge pull request #317 from Araavinds/optimise

Code optimised and added some testing

priyan-coder committed:
done

priyan-coder committed:
Merge branch 'master' into refactor-again-export

priyan-coder committed:
Merge pull request #318 from priyan-coder/refactor-again-export

Done with refactor of export command

Mudaafi committed:
Last branch (#319)

* Made some minor bugfixes.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

* Fixed height issue.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

* LAST BUGFIX

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
User inputs not corresponding to a particular CommandType/TaskType will now be ignored.

Signed-off-by: Mudaafi <SleepyUndergrad@gmail.com>

Mudaafi committed:
Added User Guide and Developer Guide

Mudaafi committed:
Updated Docs
